src/main/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/RptTransactionsServiceImpl.java

```diff
@@ -0,0 +1,106 @@
+package uk.gov.companieshouse.web.accounts.service.smallfull.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.util.UriTemplate;
+import uk.gov.companieshouse.api.ApiClient;
+import uk.gov.companieshouse.api.error.ApiErrorResponseException;
+import uk.gov.companieshouse.api.handler.exception.URIValidationException;
+import uk.gov.companieshouse.api.model.ApiResponse;
+import uk.gov.companieshouse.api.model.accounts.smallfull.relatedpartytransactions.RptTransactionApi;
+import uk.gov.companieshouse.web.accounts.api.ApiClientService;
+import uk.gov.companieshouse.web.accounts.exception.ServiceException;
+import uk.gov.companieshouse.web.accounts.model.relatedpartytransactions.AddOrRemoveRptTransactions;
+import uk.gov.companieshouse.web.accounts.model.relatedpartytransactions.RptTransaction;
+import uk.gov.companieshouse.web.accounts.model.relatedpartytransactions.RptTransactionToAdd;
+import uk.gov.companieshouse.web.accounts.service.smallfull.RptTransactionService;
+import uk.gov.companieshouse.web.accounts.transformer.smallfull.relatedpartytransactions.RptTransactionsTransformer;
+import uk.gov.companieshouse.web.accounts.util.ValidationContext;
+import uk.gov.companieshouse.web.accounts.validation.ValidationError;
+import uk.gov.companieshouse.web.accounts.validation.helper.ServiceExceptionHandler;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class RptTransactionsServiceImpl implements RptTransactionService {
+
+    @Autowired
+    private ApiClientService apiClientService;
+
+    @Autowired
+    private RptTransactionsTransformer rptTransactionsTransformer;
+
+    @Autowired
+    private ServiceExceptionHandler serviceExceptionHandler;
+
+    @Autowired
+    private ValidationContext validationContext;
+
+    private static final UriTemplate RPT_TRANSACTIONS_URI =
+            new UriTemplate("/transactions/{transactionId}/company-accounts/{companyAccountsId}/small-full/notes/related-party-transactions/transactions");
+
+    private static final UriTemplate RPT_TRANSACTIONS_URI_WITH_ID =
+            new UriTemplate("/transactions/{transactionId}/company-accounts/{companyAccountsId}/small-full/notes/related-party-transactions/transactions/{transactionId}");
+
+    private static final String RESOURCE_NAME = "transactions";
```

Comment:
Are they "transactions" or "rptTransactions" ?

Reply:
I think that, within the JSON, these are 'transactions', aren't they?

Reply:
It is a little confusing to have it sometimes be transactions, sometimes rptTransactions.

Reply:
Classes and fields clash in some of the cases so i have used a consistent approach to declare those as RptTransactions while resource name is "transactions" and that is why i have assigned it transactions string value 👍